### DISCUSSION TITLE
v0.10.0 - The one with `seed.ts`

### DISCUSSION THREAD

**Discussion by justinvdm** [snaplet/docs-old#85] [Changelog]
We [recently added](https://www.loom.com/share/e48e4c9cba4e4d92b92d37f6ddeddad7?t=4) a new feature to our snaplet cli: `snaplet seed` - a way to populate an empty database with generated data.

In this release, you're now able to configure how this seed data looks - in much the same way as how you might've [configured how snaplet transforms](https://docs.snaplet.dev/getting-started/data-operations#transforming-data) your production data before.

When you run `snaplet config setup`, or `snaplet config generate -t seed`, we'll now generate an example `.snaplet/seed.ts` config file in your project dir. It'll look something like this:

```ts
// This is an example config generated by snaplet.
// It shows how you can override the default data generated by `snaplet seed`.
// Add more columns, tables, or schemas to override this default data.
import { copycat, faker } from "@snaplet/copycat";
import type { Transform } from "./structure";

export const config: Transform = () => ({
  public: {
    User({ row }) {
      return {
        name: copycat.fullName(row.name),
      };
    },
  },
}); 
```

The file shows an example of how you can configure `seed`. In this case, `row.name` represents the value snaplet has generated for the `name` column in the `User` table for a particular row.

Lets say your `name` column was actually representing first names rather than full names. You could change it to something like this:

```ts
export const config: Transform = () => ({
  public: {
    User({ row }) {
      return {
        name: copycat.firstName(row.name)
      };
    },
  },
}); 
```

**Comment by CarelFdeWaal** [comment]
This is awesome!
